changefeedccl: TestGetCheckpointSpans failed (index out of bounds)

https://mesolite.cluster.engflow.com/invocations/default/66be725b-8bd0-4a98-85fb-6de5e2a67a8a

```
=== RUN   TestGetCheckpointSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestGetCheckpointSpans1223301965
    test_log_scope.go:81: use -show-logs to present logs inline
    panic.go:114: -- test log scope end --
ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.
test logs left over in: outputs.zip/logTestGetCheckpointSpans1223301965
    panic.go:114: panic: runtime error: index out of range [99] with length 99
--- FAIL: TestGetCheckpointSpans (0.00s)
panic: runtime error: index out of range [99] with length 99 [recovered]
	panic: runtime error: index out of range [99] with length 99 [recovered]
	panic: runtime error: index out of range [99] with length 99
goroutine 16514 [running]:
testing.tRunner.func1.2({0x6b45720, 0xc007d358a8})
	GOROOT/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1634 +0x377
panic({0x6b45720?, 0xc007d358a8?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:137 +0x38a
panic({0x6b45720?, 0xc007d358a8?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestGetCheckpointSpans(0xc008439040)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors_test.go:225 +0xa33
testing.tRunner(0xc008439040, 0x740d900)
	GOROOT/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1742 +0x390
```

Jira issue: CRDB-41911

--- a/pkg/ccl/changefeedccl/changefeed_processors_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_processors_test.go
@@ -222,7 +222,7 @@ func TestGetCheckpointSpans(t *testing.T) {
 	j := 0
 	for _, s := range spans {
 		catchupFromHWM += s.ts.GoTime().Sub(hwm.GoTime())
-		if cpSpans[j].Equal(s.span) {
+		if j < len(cpSpans) && cpSpans[j].Equal(s.span) {
 			catchupFromCheckpoint += s.ts.GoTime().Sub(cpTS.GoTime())
 			j++
 		}

